src/Futhark/Analysis/HORep/SOAC.hs

```diff
@@ -377,7 +377,7 @@ transposeInput k n inp =
 -- | A definite representation of a SOAC expression.
 data SOAC lore
   = Stream SubExp (StreamForm lore) (Lambda lore) [Input]
-  | Scatter SubExp (Lambda lore) [Input] [(SubExp, Int, VName)]
+  | Scatter SubExp (Lambda lore) [Input] [([SubExp], Int, VName)]
```

Comment:
I think this `[SubExp]` should be `Shape`, to more clearly denote its purpose.

Reply:
I'm guessing that should also be the case in Futhark.IR.SOACS.SOAC?

Reply:
Yes.